src/events/render-advert.ts

```diff
@@ -179,6 +178,21 @@ const addContainerClass = (adSlotNode: HTMLElement, isRendered: boolean) => {
 		});
 };
 
+/**
+ * Check if the ad slot has a corresponding iframe to indicte the ad has rendered.
+ * @param adSlot
+ * @returns
+ */
+const hasIframe = (adSlot: HTMLElement): Promise<boolean> =>
```

Comment:
This is a simplified version of the current function. All the IE specific code has been removed and the promise return value is now a boolean:

https://github.com/guardian/commercial/blob/cdefafe37158b6ca290a181ea3a3cc222d94ae5f/src/lib/dfp/get-ad-iframe.ts#L5-L44